### ISSUE TITLE
Discussion about the loss function

### ISSUE THREAD

**Issue by yu-changqian** [wkentaro/pytorch-fcn#7]
Hi, I did an experiment about the loss function:
```
log_p = log_p.transpose(1, 2).transpose(2, 3).contiguous().view(-1, c)
```
I find the code segment with `view(-1, c)` or without both works.Why? In my opinion, the code without `view(-1, c)` is the true one. Or, in the memory, the both are actually the same. How about your idea?

**Comment by wkentaro** [comment]
Do you mean both
`log_p = log_p.tranpose(1, 2).transpose(2, 3).contiguous()`
and
`log_p = log_p.tranpose(1, 2).transpose(2, 3).contiguous().view(-1, c)`
works?

**Comment by wkentaro** [comment]
If so, maybe it is automatically reshaped in F.nll_loss.

**Comment by yu-changqian** [comment]
Yes. They both works.
